cli/src/commands/git/fetch.rs

```diff
@@ -35,30 +35,43 @@ use crate::git_util::print_git_import_stats;
 use crate::git_util::with_remote_git_callbacks;
 use crate::ui::Ui;
 
+/// Specifies which branches to fetch from remotes
+#[derive(Clone, Debug)]
+enum BranchesToFetch<'a> {
```

Comment:
If you allow `--tracked -b` (whether as a union or as an intersection), I don't think you still need the type. Having the function take two arguments would seem to work fine. That's what I meant when I said that, if we think `--tracked -b` will potentially be allowed, then maybe the function arguments you had in the previous version of the commit work fine.

OTOH, if we decide *not* to allow `--tracked -b` in the near future, a type like this would be great, I think.